src/librustc_lint/builtin.rs

```diff
@@ -1825,6 +1825,10 @@ impl Async2018 {
         if cx.sess.parse_sess.raw_identifier_spans.borrow().contains(&span) {
             return;
         }
+        // Don't suggest about raw identifiers if the feature isn't active
+        if !cx.sess.features_untracked().raw_identifiers {
+            return
```

Comment:
We can still lint, we should just not have a suggestion in that case, but simply a note about choosing a different name.